Parking-Zone/Areas/Admin/Controllers/ParkingSlotController.cs

```diff
@@ -8,32 +8,74 @@ namespace Parking_Zone.Areas.Admin.Controllers
     [Area("Admin")]
     public class ParkingSlotController : Controller
     {
-        private readonly IParkingSlotService _parkingSlotService;
-        private readonly IParkingZoneService _parkingZoneService;
+        private readonly IParkingSlotService _slotService;
+        private readonly IParkingZoneService _zoneService;
 
-        public ParkingSlotController(IParkingZoneService parkingZoneService, IParkingSlotService parkingSlotService)
+        public ParkingSlotController(IParkingZoneService zoneService, IParkingSlotService slotService)
         {
-            _parkingSlotService = parkingSlotService;
-            _parkingZoneService = parkingZoneService;
+            _slotService = slotService;
+            _zoneService = zoneService;
         }
 
-        public IActionResult Index(Guid parkingZoneId)
+        public IActionResult Index(Guid zoneId)
         {
-            var parkingZone = _parkingZoneService.GetById(parkingZoneId);
+            var Zone = _zoneService.GetById(zoneId);
 
-            if (parkingZone is null)
+            if (Zone is null)
             {
                 return BadRequest();
             }
 
-            var slots = _parkingSlotService.GetByParkingZoneId(parkingZoneId);
+            var slots = _slotService.GetByParkingZoneId(zoneId);
 
             var slotVMs = slots.Select(x => new ParkingSlotListItemVM(x)).ToList();
 
-            ViewData["parkingZoneName"] = parkingZone.Name;
-            ViewData["parkingZoneId"] = parkingZoneId;
+            ViewData["parkingZoneName"] = Zone.Name;
+            ViewData["parkingZoneId"] = zoneId;
 
             return View(slotVMs);
         }
+
+        // GET: Admin/ParkingSlots/Create
+        public IActionResult Create(Guid zoneId)
+        {
+            var Zone = _zoneService.GetById(zoneId);
+
+            if (Zone is null)
+            {
+                return BadRequest();
+            }
+
+            var SlotCreateVM = new ParkingSlotCreateVM();
+
+            SlotCreateVM.ParkingZoneId = zoneId;
+            SlotCreateVM.ParkingZoneName = Zone.Name;
+
+            return View(SlotCreateVM);
+        }
+
+        // POST: Admin/ParkingSlots/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create(ParkingSlotCreateVM slotCreateVM)
+        {
+            if (_slotService.SlotExistsWithThisNumber(slotCreateVM.Number, null, slotCreateVM.ParkingZoneId))
+            {
+                ModelState.AddModelError("Number", "This Slot number already exists");
+            }
+
+            if (ModelState.IsValid)
+            {
+                var Zone = _zoneService.GetById(slotCreateVM.ParkingZoneId);
+
+                if (Zone is null)
+                    return BadRequest("Parking Zone Not Found");
```

Comment:
Add test for this case. See [here](https://github.com/DilshodMQ/Parking-zone/pull/9/files#diff-0be201021cb04ff536474e37b39372b678b1d2ab895950064dab90d36337ae85R228)